### PR TITLE
updated params.json with databricks resource id

### PR DESCRIPTION
**Pull request by bhavink**
this is related to deploying and initializing databricks workspace using service principals fully automated workflow

### CODE REVIEW

**Comment by bhavink** [comment]
updated get_pat with new functionality, we now generate aad token in a non interactive way

File: examples/standard/params.json [hunk 1]
```diff
@@ -12,5 +12,6 @@
         "tenant_name": "myaadtenant.onmicrosoft.com",
         "stack_config_path": "../examples/standard/stack/stack_config.json",
         "users_groups_path": "../examples/standard/users_and_groups.json",
-        "log_level": "info"
-}
+        "log_level": "info",
+        "databricks_resource_id": '2ff814a6-3304-4ab8-85cb-cd0e6f879c1d'
```
Comment (dillonbostwick-zz) [inline examples/standard/params.json]: Can we avoid adding a credential unless its a redacted example

File: src/get_pat.py [hunk 2]
```diff
@@ -2,131 +2,272 @@
...
-
-# Use Token API to exhange AAD token for PAT
-def pat_from_aad_token(aad_token, db_host, subscription_id, rg_name, ws_name):
+
+
+def get_adb_authentication_client_key(params):
+	"""
+    Authenticate using service principal w/ key.
+    """
+	authority_host_uri = 'https://login.microsoftonline.com'
```
Comment (dillonbostwick-zz) [inline src/get_pat.py]: Can we add this as a global up top so its easier to change later or parameterize if needed

File: src/get_pat.py [hunk 3]
```diff
@@ -2,131 +2,272 @@
...
+Using the bearer token (AAD access token) we cannot query the role or permission of the user/sp in the Azure Databricks Resource, 
+but using the management token we are able to query AAD for the role of the user/sp
+'''
+
+
+def get_adb_authorization_client_key(params):
+	"""
+    Authenticate using service principal w/ key.
+    """
+	authority_host_uri = 'https://login.microsoftonline.com'
```
Comment (dillonbostwick-zz) [inline src/get_pat.py]: Same here

File: src/get_pat.py [hunk 4]
```diff
@@ -2,131 +2,272 @@
...
+	tenant = params['tenant_id']
+	authority_uri = authority_host_uri + '/' + tenant
+	resource_uri = 'https://management.core.windows.net/'
```
Comment (dillonbostwick-zz) [inline src/get_pat.py]: Same here - re globals

File: src/initialize_workspace.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+
+def adb_authorization_client_key(params):
+	"""
+    Authenticate using service principal w/ key.
+    """
+	authority_host_uri = 'https://login.microsoftonline.com'
+	tenant = params['tenant_id']
+	authority_uri = authority_host_uri + '/' + tenant
+	resource_uri = 'https://management.core.windows.net/'
```
Comment (dillonbostwick-zz) [inline src/initialize_workspace.py:46]: Change as global (in case we need to parameterize later)

**Review by dillonbostwick-zz — CHANGES_REQUESTED** [review]
Thanks for making the changes.

Feel free to remove the commented out old code that I had. It will be in commit history if we need it later.

**Comment by bhavink** [comment]
removed commented section in get_pat also updated setup.py with dependecies

File: examples/standard/params.json (hunk 1 above)
Comment (bhavink) [reply examples/standard/params.json]: done, removed comments

File: src/get_pat.py (hunk 2 above)
Comment (bhavink) [reply src/get_pat.py]: done, updated code with requested changes

File: src/get_pat.py (hunk 3 above)
Comment (bhavink) [reply src/get_pat.py]: done

File: src/get_pat.py (hunk 4 above)
Comment (bhavink) [reply src/get_pat.py]: done

File: src/initialize_workspace.py (hunk 5 above)
Comment (bhavink) [reply src/initialize_workspace.py:46]: done

File: examples/standard/params.json (hunk 1 above)
Comment (bhavink) [reply examples/standard/params.json]: this is databricks resource id, we can remove it, its not needed anymore

**Comment by bhavink** [comment]
implemented all of the requested changes

**Review by dillonbostwick-zz — APPROVED** [review]

**Pull request merged**